katz/src/main/kotlin/katz/data/Ior.kt

```diff
@@ -266,17 +266,17 @@ sealed class Ior<out A, out B> {
  *
  * @param f The function to bind across [Ior.Right].
  */
-inline fun <A, B, reified C : A, D> Ior<A, B>.flatMap(aa: Semigroup<C>, crossinline f: (B) -> Ior<C, D>): Ior<C, D> = when (this) {
-    is Ior.Left -> Ior.Left(value as C)
+inline fun <A, B, D> Ior<A, B>.flatMap(AA: Semigroup<A>, crossinline f: (B) -> Ior<A, D>): Ior<A, D> = when (this) {
```

Comment:
This solves a problem on #48. I'll revisit it after this gets merged :D

Could you please use a name that's lowercase?